tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/evaluation/ExpressionEvaluationTest.java

```diff
@@ -0,0 +1,283 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.debugger.test.adapter.evaluation;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.ballerinalang.debugger.test.DebugAdapterBaseTestCase;
+import org.ballerinalang.debugger.test.utils.BallerinaTestDebugPoint;
+import org.ballerinalang.debugger.test.utils.DebugUtils;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.eclipse.lsp4j.debug.StoppedEventArguments;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+/**
+ * Test implementation for debug expression evaluation scenarios.
+ */
+public class ExpressionEvaluationTest extends DebugAdapterBaseTestCase {
+
+    private StoppedEventArguments context;
+
+    private static final String nilVar = "v01_varVariable";
+    private static final String booleanVar = "v02_booleanVar";
+    private static final String intVar = "v03_intVar";
+    private static final String floatVar = "v04_floatVar";
+    private static final String decimalVar = "v05_decimalVar";
+    private static final String stringVar = "v06_stringVar";
+    private static final String xmlVar = "v07_xmlVar";
+    private static final String arrayVar = "v08_arrayVar";
+    private static final String tupleVar = "v09_tupleVar";
+    private static final String mapVar = "v10_mapVar";
+    private static final String recordVar = "v11_john";
+    private static final String anonRecordVar = "v12_anonRecord";
+    private static final String errorVar = "v13_errorVar";
+    private static final String anonFunctionVar = "v14_anonFunctionVar";
+    private static final String futureVar = "v15_futureVar";
+    private static final String objectVar = "v16_objectVar";
+    private static final String anonObjectVar = "v17_anonObjectVar";
+    private static final String typeDescVar = "v18_typedescVar";
+    private static final String unionVar = "v19_unionVar";
+    private static final String optionalVar = "v20_optionalVar";
+    private static final String anyVar = "v21_anyVar";
+    private static final String anydataVar = "v22_anydataVar";
+    private static final String byteVar = "v23_byteVar";
+    private static final String jsonVar = "v24_jsonVar";
+    private static final String tableVar = "v25_tableVar";
+    private static final String streamVar = "v26_oddNumberStream";
+    private static final String neverVar = "v27_neverVar";
+
+    @BeforeClass
+    public void setup() throws BallerinaTestException {
+        testProjectName = "basic-project";
+        testModuleName = "advanced";
+        testModuleFileName = "main.bal";
+        testProjectPath = testProjectBaseDir.toString() + File.separator + testProjectName;
```

Comment:
Shall we change the above lines to use `Paths.get` since `File.sperator` is a pre-1.7 JDK way?
```suggestion
        testProjectPath = Paths.get(testProjectBaseDir.toString(), testProjectName).toString();
```

Reply:
+1, Changed.